oneargopy/Argo.py

```diff
@@ -0,0 +1,279 @@
+
+                    # Remove extraneous .gz file
+                    gz_save_path.unlink()
+                    
+                    # Exit the loop if download is successful so we don't try additional
+                    # sources for no reason.
+                    break 
+
+                except requests.RequestException as e:
+                    print(f'Error encountered: {e}. Trying next host...')
+            
+            # Increment Iterations
+            iterations += 1
+
+        # If ultimately nothing could be downloaded
+        if not success: 
+            if update_status:
+                print(f'WARNING: Update for {file_name} failed, you are working with out of date data.')
+            else:
+             raise Exception(f'Download failed! {file_name} could not be downloaded at this time.')
+        
+
+    def __load_synthetic_dataframe(self) -> pd:
+        """ A function to load an index file into a data frame for easier reference.
+
+            :param: file_name : str - The name of the file that we would like
+                to read into a dataframe.
+
+            NOTE: the header is 8 here because there are 8 lines in both index fiels
+                devoted to header information.
+            NOTE: R: raw data, A: adjusted mode (real-time adjusted), D: delayed mode quality controlled
+
+        """
+        file_name = "argo_synthetic-profile_index.txt"
+        file_path = Path.joinpath(self.download_settings.base_dir, 'Index', file_name)
+        synthetic_index = pd.read_csv(file_path, delimiter=',', header=8, parse_dates=['date','date_update'], 
+                                date_format='%Y%m%d%H%M%S')
+        
+        # Parsing out variables in first column file
+        dacs = synthetic_index['file'].str.split('/').str[0]
+        synthetic_index.insert(0, "dacs", dacs, True)
+
+        wmoid = synthetic_index['file'].str.split('/').str[1]
+        synthetic_index.insert(1, "wmoid", wmoid, True)
+
+        profile = synthetic_index['file'].str.split('_').str[1].str.replace('.nc', '')
+        synthetic_index.insert(2, "profile", profile, True)
+
+        # Splitting the parameters into their own collumns
+        parameters_split = synthetic_index['parameters'].str.split()
+        data_types_split = synthetic_index['parameter_data_mode'].apply(list)
+
+        data_type_mapping = {np.nan: 0, 'R':1, 'A':2, 'D':3 }
+        mapped_data_types_split = data_types_split.apply(lambda lst: [data_type_mapping.get(x, 0) if pd.notna(x) else 0 for x in lst])
+
+        # Create a new DataFrame from the split lists
+        expanded_df = pd.DataFrame({
+            'index': synthetic_index.index.repeat(parameters_split.str.len()),
+            'parameter': parameters_split.explode(),
+            'data_type': mapped_data_types_split.explode()
+        })
+
+        # Pivot the expanded DataFrame to get parameters as columns
+        with pd.option_context('future.no_silent_downcasting', True):
```

Comment:
future.no_silent_downcasting requires pandas 2.2.2, which I don't have
Please comment out for now